src/middlewared/middlewared/plugins/account.py

```diff
@@ -873,6 +874,9 @@ def do_delete(self, audit_callback, pk, options):
         user = self.middleware.call_sync('user.get_instance', pk)
         audit_callback(user['username'])
 
+        if app and user['username'] == app.authenticated_credentials.user['username']:
```

Comment:
You need to check whether it's a user session as well. See examples in `utils/privilege.py` 
app and app.authenticated_credentials.is_user_session.